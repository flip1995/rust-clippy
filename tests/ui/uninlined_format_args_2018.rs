// run-rustfix
// edition:2018

#![warn(clippy::uninlined_format_args)]

fn main() {
    let var = 1;

    println!("val='{}'", var);

    if var > 0 {
        panic!("p1 {}", var);
    }
    if var > 0 {
        panic!("p2 {0}", var);
    }
    if var > 0 {
        panic!("p3 {var}", var = var);
    }

    #[allow(non_fmt_panics)]
    {
        if var > 0 {
            panic!("p4 {var}");
        }
    }
}
