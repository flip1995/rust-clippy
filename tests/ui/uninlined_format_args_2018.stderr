error: variables can be used directly in the `format!` string
  --> $DIR/uninlined_format_args_2018.rs:9:5
   |
LL |     println!("val='{}'", var);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::uninlined-format-args` implied by `-D warnings`
help: change this to
   |
LL -     println!("val='{}'", var);
LL +     println!("val='{var}'");
   |

error: aborting due to previous error

